src/main/java/org/openscience/cdk/geometry/cip/CIPToolMod.java

```diff
@@ -271,8 +296,61 @@ private static int permParity(final ILigand[] ligands) {
         return (swaps & 0x1) == 0x1 ? -1 : +1;
     
     }
-    
 
-    
-   
+    public static int getSizeOfLargestRingSystem(CdkChemicalImpl chemical) {
+        CdkChemicalImpl copy = chemical.deepCopy();
+        int maxRings = 0;
+        for(int i = copy.getBondCount()-1; i >=0; i--) {
+            Bond bond = copy.getBond(i);
+            if(!bond.isInRing() ) {
+                copy.removeBond(i);
+            }
+        }
+
```

Comment:
If we didn't find a ring bond by here, we should just return 0 I think, save some unnecessary processing.